plugin/effect/src/index.ts

```diff
@@ -1,21 +1,20 @@
-import { Effect } from "effect";
+import * as Effect from "effect/Effect";
 import * as z from "zod";
 
 function zodEffect(this: z.ZodType, data: unknown, params?: any) {
-  return Effect.tryPromise({
-    try: async () => this.parseAsync(data, params),
-    catch(error) {
-      return error as z.ZodError;
-    },
-  });
+  return Effect.flatMap(
+    Effect.promise(() => this.safeParseAsync(data, params)),
+    (result) =>
+      result.success ? Effect.succeed(result.data) : Effect.fail(result.error)
+  );
 }
 
 function zodEffectSync(this: z.ZodType, data: unknown, params?: any) {
-  return Effect.try({
-    try: () => this.parse(data, params),
-    catch(error) {
-      return error as z.ZodError;
-    },
+  return Effect.suspend(() => {
```

Comment:
Effect.suspend allows us to wrap a side effect that returns an Effect, perfect for the sync api.